list_ends.py

```diff
@@ -13,8 +14,20 @@ def generate(length):
         l.append(random.randrange(elem_range))
     return l
 
-l = generate(random.randrange(1, max_length))
+try:
+    len = int(input("Rozmiar populacji do utworzenia listy (dla <=0 generuj w domyslny sposob): "))
+    if len > 0:
+        n = int(input("Rozmiar listy: "))
+        l = random.sample(range(len), n)
```

Comment:
Mała uwaga ...

Twoja funkcja generate losuje próbkę z zwracaniem.

Żeby to samo uzyskać bezpośrednio z pakietu random zamiast random.sample można użyć random.choices